<a href="https://colab.research.google.com/github/Hemanath17/finedatanalysis/blob/main/Data_Analysis_colab.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [7]:
#import libraries
import pandas as pd
import seaborn as sns
import numpy as np
from google.colab import files
files.upload()

Saving fines.csv to fines (1).csv


{'fines.csv': b'org_indiv,first_name,last_name,city,state,category,cash_fine,online_fine\r\n3-D MEDICAL SERVICES LLC,STEVEN,DEITELZWEIG,NEW ORLEANS,LA,Professional Advising,"2,625.00",-\r\n"AA DOCTORS, INC.",AAKASH,AHUJA,PASO ROBLES,CA,Expert-Led Forums,"1,000.00",-\r\n"ABBO, LILIAN MARGARITA",LILIAN,ABBO,MIAMI,FL,Business Related Travel,-,448\r\n"ABBO, LILIAN MARGARITA",LILIAN,ABBO,MIAMI,FL,Meals,-,119\r\n"ABBO, LILIAN MARGARITA",LILIAN,ABBO,MIAMI,FL,Professional Advising,"1,800.00",-\r\nABDULLAH RAFFEE MD PC,ABDULLAH,RAFFEE,FLINT,MI,Expert-Led Forums,750,-\r\n"ABEBE, SHEILA Y",SHEILA,ABEBE,INDIANAPOLIS,IN,Educational Items,-,47\r\n"ABEBE, SHEILA Y",SHEILA,ABEBE,INDIANAPOLIS,IN,Expert-Led Forums,825,-\r\nABILENE FAMILY FOOT CENTER,GALEN,ALBRITTON,ABILENE,TX,Professional Advising,"3,000.00",-\r\n"ABOLNIK, IGOR Z",IGOR,ABOLNIK,PROVO,UT,Business Related Travel,-,396\r\n"ABOLNIK, IGOR Z",IGOR,ABOLNIK,PROVO,UT,Expert-Led Forums,"1,750.00",-\r\n"ABOLNIK, IGOR Z",IGOR,ABOLNIK,PROVO,UT,Meals,

In [8]:
from google.colab import files
files.upload()

Saving Emails sent-sheet.csv to Emails sent-sheet (1).csv


{'Emails sent-sheet.csv': b"name_last,name_first,name_middle,mail_sent_date,office\r\nADELGLASS,JEFFREY,M.,2009-05-22 00:00:00,Center for Drug Evaluation and Research\r\nADKINSON,N.,FRANKLIN, 2010-04-17,Center for Biologics Evaluation and Research\r\nALLEN,MARK,S.,2012-01-26,Center for Devices and Radiological Health\r\nAMSTERDAM,DANIEL,,2014.11.15,Center for Biologics Evaluation and Research\r\nAMSTUTZ,HARLAN,C.,2014.07.17,Center for Devices and Radiological Health\r\nANDERSON,C.,JOSEPH, 2010-02-22,Center for Devices and Radiological Health\r\nANDREWS,DAVID,W., 2010-07-17,Center for Biologics Evaluation and Research\r\nAQEL,RAED,,2012-10-27,Center for Devices and Radiological Health\r\nARROWSMITH,PETER,N.,2014.01.18,Center for Devices and Radiological Health\r\nBARR,JOHN,D., 2010-01-11,Center for Devices and Radiological Health\r\nBARTHOLOMEW,BRADLEY,J.,2016-11-05 00:00:00,Center for Devices and Radiological Health\r\nBATSHAW,MARK,L., 2010-11-28,Center for Biologics Evaluation and Res

In [9]:
#load fines paid by doctors dataset
fine_df = pd.read_csv("fines.csv")
fine_df.head()

org_indiv first_name  ... cash_fine online_fine
0  3-D MEDICAL SERVICES LLC     STEVEN  ...  2,625.00           -
1          AA DOCTORS, INC.     AAKASH  ...  1,000.00           -
2    ABBO, LILIAN MARGARITA     LILIAN  ...         -         448
3    ABBO, LILIAN MARGARITA     LILIAN  ...         -         119
4    ABBO, LILIAN MARGARITA     LILIAN  ...  1,800.00           -

[5 rows x 8 columns]

remove the $ symbol from cash_fine and online_fine since it is not necessary for the data analysis

In [10]:
#load emails sent dataset
email_df = pd.read_csv("Emails sent-sheet.csv")
email_df.head()

name_last  ...                                        office
0  ADELGLASS  ...       Center for Drug Evaluation and Research
1   ADKINSON  ...  Center for Biologics Evaluation and Research
2      ALLEN  ...    Center for Devices and Radiological Health
3  AMSTERDAM  ...  Center for Biologics Evaluation and Research
4    AMSTUTZ  ...    Center for Devices and Radiological Health

[5 rows x 5 columns]

In [11]:
#print info of data
fine_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10087 entries, 0 to 10086
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   org_indiv    10087 non-null  object
 1   first_name   9884 non-null   object
 2   last_name    10087 non-null  object
 3   city         10087 non-null  object
 4   state        10087 non-null  object
 5   category     10086 non-null  object
 6   cash_fine    10086 non-null  object
 7   online_fine  10084 non-null  object
dtypes: object(8)
memory usage: 630.6+ KB


In [12]:
#find null count
fine_df.isnull().sum()

org_indiv        0
first_name     203
last_name        0
city             0
state            0
category         1
cash_fine        1
online_fine      3
dtype: int64

In [13]:
fine_df["category"].fillna("Not mentioned", inplace=True)

In [14]:
#replace - with 0
fine_df["cash_fine"].fillna("0", inplace=True)
fine_df["online_fine"].fillna("0", inplace=True)

In [15]:
fine_df.isnull().sum()

org_indiv        0
first_name     203
last_name        0
city             0
state            0
category         0
cash_fine        0
online_fine      0
dtype: int64

In [16]:
fine_df["first_name"].fillna("", inplace = True)

In [17]:
fine_df.isnull().sum()

org_indiv      0
first_name     0
last_name      0
city           0
state          0
category       0
cash_fine      0
online_fine    0
dtype: int64

In [18]:
fine_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10087 entries, 0 to 10086
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   org_indiv    10087 non-null  object
 1   first_name   10087 non-null  object
 2   last_name    10087 non-null  object
 3   city         10087 non-null  object
 4   state        10087 non-null  object
 5   category     10087 non-null  object
 6   cash_fine    10087 non-null  object
 7   online_fine  10087 non-null  object
dtypes: object(8)
memory usage: 630.6+ KB


In [19]:
fine_df.describe()

org_indiv  ... online_fine
count                                     10087  ...       10087
unique                                     4851  ...        1267
top     REGENTS OF THE UNIVERSITY OF CALIFORNIA  ...           -
freq                                         10  ...        4661

[4 rows x 8 columns]

In [20]:
fine_df['cash_fine'].replace({'-':'0'},inplace=True)
fine_df['online_fine'].replace({'-':'0'},inplace=True)

In [21]:
fine_df.head()

org_indiv first_name  ... cash_fine online_fine
0  3-D MEDICAL SERVICES LLC     STEVEN  ...  2,625.00           0
1          AA DOCTORS, INC.     AAKASH  ...  1,000.00           0
2    ABBO, LILIAN MARGARITA     LILIAN  ...         0         448
3    ABBO, LILIAN MARGARITA     LILIAN  ...         0         119
4    ABBO, LILIAN MARGARITA     LILIAN  ...  1,800.00           0

[5 rows x 8 columns]

In [22]:
email_df.head()

name_last  ...                                        office
0  ADELGLASS  ...       Center for Drug Evaluation and Research
1   ADKINSON  ...  Center for Biologics Evaluation and Research
2      ALLEN  ...    Center for Devices and Radiological Health
3  AMSTERDAM  ...  Center for Biologics Evaluation and Research
4    AMSTUTZ  ...    Center for Devices and Radiological Health

[5 rows x 5 columns]

## Changing mail_sent_date in standardized format

In [23]:
#remove the time format from the mail_sent_date
email_df['mail_sent_date']=pd.to_datetime(email_df['mail_sent_date'].astype(str), format='%Y-%m-%d')
email_df.head()

name_last  ...                                        office
0  ADELGLASS  ...       Center for Drug Evaluation and Research
1   ADKINSON  ...  Center for Biologics Evaluation and Research
2      ALLEN  ...    Center for Devices and Radiological Health
3  AMSTERDAM  ...  Center for Biologics Evaluation and Research
4    AMSTUTZ  ...    Center for Devices and Radiological Health

[5 rows x 5 columns]

In [24]:
email_df.isnull().sum()

name_last          0
name_first         0
name_middle       91
mail_sent_date     0
office             0
dtype: int64

In [25]:
#changing the date into standardized date format and then creating a new column for year
import datetime
email_df['mail_sent_date'] = pd.to_datetime(email_df['mail_sent_date'])
email_df['year'] = email_df['mail_sent_date'].dt.year
email_df['year'][1].astype(int)
email_df.head()

name_last name_first  ...                                        office  year
0  ADELGLASS    JEFFREY  ...       Center for Drug Evaluation and Research  2009
1   ADKINSON         N.  ...  Center for Biologics Evaluation and Research  2010
2      ALLEN       MARK  ...    Center for Devices and Radiological Health  2012
3  AMSTERDAM     DANIEL  ...  Center for Biologics Evaluation and Research  2014
4    AMSTUTZ     HARLAN  ...    Center for Devices and Radiological Health  2014

[5 rows x 6 columns]

In [26]:
email_df['mail_sent_date'].head(10)

0   2009-05-22
1   2010-04-17
2   2012-01-26
3   2014-11-15
4   2014-07-17
5   2010-02-22
6   2010-07-17
7   2012-10-27
8   2014-01-18
9   2010-01-11
Name: mail_sent_date, dtype: datetime64[ns]

In [27]:
#check for null values
email_df.isnull().sum()

name_last          0
name_first         0
name_middle       91
mail_sent_date     0
office             0
year               0
dtype: int64

In [28]:
email_df["name_middle"].fillna("", inplace = True)
#we can either fill the middle name with either space or NaN since it is not necessary

In [29]:
email_df.isnull().sum()

name_last         0
name_first        0
name_middle       0
mail_sent_date    0
office            0
year              0
dtype: int64

In [30]:
email_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 272 entries, 0 to 271
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   name_last       272 non-null    object        
 1   name_first      272 non-null    object        
 2   name_middle     272 non-null    object        
 3   mail_sent_date  272 non-null    datetime64[ns]
 4   office          272 non-null    object        
 5   year            272 non-null    int64         
dtypes: datetime64[ns](1), int64(1), object(4)
memory usage: 12.9+ KB


In [31]:
email_df['year'].describe()

count     272.000000
mean     2013.694853
std         3.583311
min      2006.000000
25%      2011.000000
50%      2014.000000
75%      2016.000000
max      2020.000000
Name: year, dtype: float64

In [32]:
#print out the unique year
email_df.year.unique()

array([2009, 2010, 2012, 2014, 2016, 2007, 2015, 2008, 2018, 2019, 2011,
       2013, 2017, 2020, 2006])

In [33]:
print("Dataset starts with the year ",email_df.year.min())

Dataset starts with the year  2006


Hence mail was being sent from the year 2006

In [34]:
print("Dataset ends with the year ",email_df.year.max())

Dataset ends with the year  2020


Hence the last mail was sent in the year 2020

In [35]:
email_df['year'].count()

272

In [36]:
year_wise = email_df.groupby(['year']).size()
print("Email sent on each year")
print(year_wise)

Email sent on each year
year
2006     1
2007    15
2008     8
2009    14
2010    25
2011    20
2012    16
2013    21
2014    32
2015    31
2016    25
2017    11
2018    23
2019    25
2020     5
dtype: int64


# Visualization of Email sent each year

In [37]:
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go
import warnings
warnings.simplefilter("ignore")

fig = px.bar(email_df, x=email_df['year'].value_counts().keys(), y=email_df['year'].value_counts(), color=email_df['year'].value_counts().keys(),
             labels={
                     'x': "Year",
                     'y': "Number of emails"
                     
                 })
fig.update_layout(title = 'Email sent on each year')
fig.show()

In [38]:
max_emails = year_wise.sort_values(ascending=False)
max_emails

year
2014    32
2015    31
2019    25
2016    25
2010    25
2018    23
2013    21
2011    20
2012    16
2007    15
2009    14
2017    11
2008     8
2020     5
2006     1
dtype: int64

From the above visualization we can conclude that in year 2014 maximum emails have been sent to the doctors

In [39]:
##########

In [40]:
import datetime
email_df['mail_sent_date'] = pd.to_datetime(email_df['mail_sent_date'])
email_df['month'] = email_df['mail_sent_date'].dt.month
email_df['month'][1].astype(int)
email_df.head()

name_last name_first  ...  year month
0  ADELGLASS    JEFFREY  ...  2009     5
1   ADKINSON         N.  ...  2010     4
2      ALLEN       MARK  ...  2012     1
3  AMSTERDAM     DANIEL  ...  2014    11
4    AMSTUTZ     HARLAN  ...  2014     7

[5 rows x 7 columns]

In [41]:
month = email_df.month.unique()
print(np.sort(month))

[ 1  2  3  4  5  6  7  8  9 10 11 12]


In [42]:
result = email_df.groupby(['month']).size()
print("month wise email sent")
print(result)

month wise email sent
month
1     19
2     19
3     28
4     21
5     23
6     28
7     35
8     12
9     19
10    26
11    27
12    15
dtype: int64


In [43]:
month_df = pd.DataFrame(result, columns = ['times'])
month_df

times
month       
1         19
2         19
3         28
4         21
5         23
6         28
7         35
8         12
9         19
10        26
11        27
12        15

In [44]:
mon = ['Jan', 'Feb', 'Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']
month_df['month'] = mon

In [45]:
month_df

times month
month             
1         19   Jan
2         19   Feb
3         28   Mar
4         21   Apr
5         23   May
6         28   Jun
7         35   Jul
8         12   Aug
9         19   Sep
10        26   Oct
11        27   Nov
12        15   Dec

In [46]:
j = month_df.times[1]
f = month_df.times[2]
m = month_df.times[3]
a = month_df.times[4]
ma = month_df.times[5]
ju= month_df.times[6]
jul = month_df.times[7]
au = month_df.times[8]
s = month_df.times[9]
o = month_df.times[10]
n = month_df.times[11]
d = month_df.times[12]

In [48]:
#import plotly for pie chart
import plotly.express as px
y = [j,f,m,a,ma,ju,jul,au,s,o,n,d]
x = ["Jan", 'Feb','Mar','Apr','May','Jun','Jul','Aug','Sep','Oct','Nov','Dec']
fig = px.pie(month_df,
             values=y, 
             names=x)
fig.update_layout(
    title='Email send each month'
)
fig.show()

In [49]:
fine_df.head()

org_indiv first_name  ... cash_fine online_fine
0  3-D MEDICAL SERVICES LLC     STEVEN  ...  2,625.00           0
1          AA DOCTORS, INC.     AAKASH  ...  1,000.00           0
2    ABBO, LILIAN MARGARITA     LILIAN  ...         0         448
3    ABBO, LILIAN MARGARITA     LILIAN  ...         0         119
4    ABBO, LILIAN MARGARITA     LILIAN  ...  1,800.00           0

[5 rows x 8 columns]

In [50]:
city = fine_df.city.unique()
city

array(['NEW ORLEANS', 'PASO ROBLES', 'MIAMI', ..., 'BATH', 'GLENCOE',
       'ROSEDALE'], dtype=object)

In [51]:
count = np.count_nonzero(city)
print(count)

1318


Hence there are totally 1318 cities from United States in the dataset

In [52]:
states = fine_df.state.unique()
states

array(['LA', 'CA', 'FL', 'MI', 'IN', 'TX', 'UT', 'OH', 'NY', 'MN', 'PR',
       'MA', 'OR', 'NJ', 'PA', 'IL', 'OK', 'TN', 'KS', 'GA', 'AR', 'WV',
       'NC', 'CT', 'DC', 'KY', 'AL', 'MD', 'WI', 'NM', 'RI', 'CO', 'SC',
       'AZ', 'ID', 'MO', 'NE', 'NV', 'MT', 'WA', 'MS', 'IA', 'NH', 'WY',
       'VA', 'DE', 'SD', 'HI', 'ME', 'VT', 'AK', 'ND'], dtype=object)

In [53]:
s_count = np.count_nonzero(states)
print(s_count)

52


The above output confirms that there were 52 states mentioned in the data

In [54]:
#Number of times fine recorded in each state
states_ = fine_df.groupby('state').count()

In [55]:
s = states_['org_indiv']
s

state
AK       1
AL     132
AR      77
AZ     154
CA    1172
CO     148
CT     155
DC      59
DE      35
FL     738
GA     248
HI      29
IA      63
ID      22
IL     448
IN     196
KS      99
KY     128
LA     145
MA     259
MD     196
ME      20
MI     342
MN     124
MO     237
MS      76
MT       9
NC     383
ND      10
NE      46
NH      27
NJ     384
NM      25
NV      56
NY     833
OH     462
OK      82
OR      58
PA     533
PR      88
RI      59
SC     149
SD      23
TN     281
TX     624
UT      84
VA     177
VT      10
WA     116
WI     169
WV      71
WY      25
Name: org_indiv, dtype: int64

In [56]:
s = s.sort_values(ascending=False)
print("Top most number of times fined in each state")
s

Top most number of times fined in each state


state
CA    1172
NY     833
FL     738
TX     624
PA     533
OH     462
IL     448
NJ     384
NC     383
MI     342
TN     281
MA     259
GA     248
MO     237
MD     196
IN     196
VA     177
WI     169
CT     155
AZ     154
SC     149
CO     148
LA     145
AL     132
KY     128
MN     124
WA     116
KS      99
PR      88
UT      84
OK      82
AR      77
MS      76
WV      71
IA      63
DC      59
RI      59
OR      58
NV      56
NE      46
DE      35
HI      29
NH      27
WY      25
NM      25
SD      23
ID      22
ME      20
ND      10
VT      10
MT       9
AK       1
Name: org_indiv, dtype: int64

From the above result it is clear that California is the most number of times occured in the dataset

In [57]:
#print out top 5
print("Top 5 cities were most number of times fine collected from the dataset")
s[:5]

Top 5 cities were most number of times fine collected from the dataset


state
CA    1172
NY     833
FL     738
TX     624
PA     533
Name: org_indiv, dtype: int64

In [58]:
#print out top 5
print("Top 5 cities were least number of times fine collected from the dataset")
s[-5:]

Top 5 cities were least number of times fine collected from the dataset


state
ME    20
ND    10
VT    10
MT     9
AK     1
Name: org_indiv, dtype: int64

In [59]:
#visulization
import plotly.graph_objects as go
sat = ['CA', 'NY','FL','TX', 'PA']
v = [1172, 833, 738, 624, 533]
fig = px.bar(s, x=sat[:5], y=v[:5], color=sat[:5],
             labels={
                     'x': "states",
                     'y': "Number of times fine collected"
                     
                 })
fig.update_layout(title = 'Top 5 cities were most number of times fine collected')
fig.show()

In [60]:
sat = ['MA', 'ND','VT','MT', 'AK']
v = [20, 10, 10, 9, 1]
fig = px.bar(s, x=sat[:5], y=v[:5], color=sat[:5],
             labels={
                     'x': "states",
                     'y': "Number of times fine collected"
                     
                 })
fig.update_layout(title = 'Top 5 cities were least number of times fine collected')
fig.show()

In [61]:
#list unique category
unique_category = fine_df.category.unique()
unique_category

array(['Professional Advising', 'Expert-Led Forums',
       'Business Related Travel', 'Meals', 'Educational Items',
       'Infosys Sponsored Research initiated before July 1, 2009',
       'Infosys Sponsored Research initiated on or after July 1, 2009',
       'Investigator-Initiated Research', 'Not mentioned'], dtype=object)

In [62]:
category = fine_df.groupby('category').count()
c = category['org_indiv']
c

category
Business Related Travel                                          2152
Educational Items                                                 171
Expert-Led Forums                                                2857
Infosys Sponsored Research initiated before July 1, 2009          203
Infosys Sponsored Research initiated on or after July 1, 2009      44
Investigator-Initiated Research                                    34
Meals                                                            3100
Not mentioned                                                       1
Professional Advising                                            1525
Name: org_indiv, dtype: int64

In [63]:
c = c.sort_values(ascending=False)
print("Most number of times fine collected on each category")
c

Most number of times fine collected on each category


category
Meals                                                            3100
Expert-Led Forums                                                2857
Business Related Travel                                          2152
Professional Advising                                            1525
Infosys Sponsored Research initiated before July 1, 2009          203
Educational Items                                                 171
Infosys Sponsored Research initiated on or after July 1, 2009      44
Investigator-Initiated Research                                    34
Not mentioned                                                       1
Name: org_indiv, dtype: int64

In [64]:
c[:5]

category
Meals                                                       3100
Expert-Led Forums                                           2857
Business Related Travel                                     2152
Professional Advising                                       1525
Infosys Sponsored Research initiated before July 1, 2009     203
Name: org_indiv, dtype: int64

In [65]:
import plotly.graph_objects as go
cat = ['Meals', 'Expert-Led Forums','Business Related Travel','Professional Advising', 'Infosys Sponsored Research initiated before July 1, 2009']
v = [3100, 2857, 2152,1525, 203]
fig = go.Figure([go.Bar(x=cat, y=[3100, 2857, 2152,1525, 203] )])
fig.show()

In [66]:
fig = px.bar(c, x=cat[:5], y=v[:5], color=cat[:5],
             labels={
                     'x': "category",
                     'y': "Number of times fine collected"
                     
                 })
fig.update_layout(title = 'Fine collected based on category')
fig.show()

In [67]:
c_count = np.count_nonzero(c)
print("Total number of categories for which the fine was collected")
print(c_count)

Total number of categories for which the fine was collected
9


In [68]:
fine_df['cash_fine'].describe()

count     10087
unique      759
top           0
freq       5424
Name: cash_fine, dtype: object

In [69]:
email_df.head()

name_last name_first  ...  year month
0  ADELGLASS    JEFFREY  ...  2009     5
1   ADKINSON         N.  ...  2010     4
2      ALLEN       MARK  ...  2012     1
3  AMSTERDAM     DANIEL  ...  2014    11
4    AMSTUTZ     HARLAN  ...  2014     7

[5 rows x 7 columns]

In [70]:
office_unique = email_df.office.unique()
office_unique

array(['Center for Drug Evaluation and Research',
       'Center for Biologics Evaluation and Research',
       'Center for Devices and Radiological Health',
       'New York District Office', 'Dallas District Office',
       'Center for Veterinary Medicine', 'Baltimore District Office',
       'New England District Office'], dtype=object)

In [71]:
office_count = np.count_nonzero(office_unique)
print("Total number of office recorded in the given data")
office_count

Total number of office recorded in the given data


8

In [72]:
office = email_df.groupby('office').count()
#count the occurence of each office in dataset to fine number of times each office was mentioned

In [73]:
office_value = office['mail_sent_date']
office_value

office
Baltimore District Office                         1
Center for Biologics Evaluation and Research     61
Center for Devices and Radiological Health      138
Center for Drug Evaluation and Research          68
Center for Veterinary Medicine                    1
Dallas District Office                            1
New England District Office                       1
New York District Office                          1
Name: mail_sent_date, dtype: int64

In [74]:
office_value_sorted = office_value.sort_values(ascending=False)
office_value_sorted

office
Center for Devices and Radiological Health      138
Center for Drug Evaluation and Research          68
Center for Biologics Evaluation and Research     61
New York District Office                          1
New England District Office                       1
Dallas District Office                            1
Center for Veterinary Medicine                    1
Baltimore District Office                         1
Name: mail_sent_date, dtype: int64

In [75]:
import plotly.graph_objects as go
off = ['Center for Devices and Radiological Health', 'Center for Drug Evaluation and Research','Center for Biologics Evaluation and Research','Baltimore District Office', 'Center for Veterinary Medicine','Dallas District Office', 'New England District Office', 'New York District Office']
val = [138, 68, 61, 1, 1, 1, 1, 1]
fig = go.Figure([go.Bar(x=off, y=val)])
fig.show()

In [76]:
#top most office recorded for fine
fig = px.bar(office_value_sorted, x=off[:5], y=val[:5], color=off[:5],
             labels={
                     'x': "name of office",
                     'y': "number of times recorded"
                     
                 })
fig.update_layout(title = 'Top 5 office recorded from the dataset')
fig.show()

In [77]:
fine_df.head(10)

org_indiv first_name  ... cash_fine online_fine
0    3-D MEDICAL SERVICES LLC     STEVEN  ...  2,625.00           0
1            AA DOCTORS, INC.     AAKASH  ...  1,000.00           0
2      ABBO, LILIAN MARGARITA     LILIAN  ...         0         448
3      ABBO, LILIAN MARGARITA     LILIAN  ...         0         119
4      ABBO, LILIAN MARGARITA     LILIAN  ...  1,800.00           0
5       ABDULLAH RAFFEE MD PC   ABDULLAH  ...       750           0
6             ABEBE, SHEILA Y     SHEILA  ...         0          47
7             ABEBE, SHEILA Y     SHEILA  ...       825           0
8  ABILENE FAMILY FOOT CENTER      GALEN  ...  3,000.00           0
9             ABOLNIK, IGOR Z       IGOR  ...         0         396

[10 rows x 8 columns]

In [78]:
online = 0
cash = 0
for i in range(0, 10087):
    if fine_df['cash_fine'][i] != '0':
        cash = cash+1
    else:
        online = online+1
        
print("Number of times an individual paid via cash:", cash)
print("Number of times an individual paid via online:", online)

Number of times an individual paid via cash: 4663
Number of times an individual paid via online: 5424


In [79]:
plot = ['Paid via cash', 'paid via online']
fig = go.Figure([go.Bar(x=plot, y=[cash, online])])
fig.show()

From this we can conclude most people prefer online fine payment that via cash payment

In [80]:
ny_online= 0
ny_cash = 0
for i in range(0, 10087):
    if fine_df['state'][i] == "NY":
        #ny_cash += 1
        if fine_df['cash_fine'][i] != '0':
            ny_cash += 1
        else:
            ny_online += 1
            
            
ca_online= 0
ca_cash = 0
for i in range(0, 10087):
    if fine_df['state'][i] == "CA":
        if fine_df['cash_fine'][i] != '0':
            ca_cash += 1
        else:
            ca_online += 1


fl_online= 0
fl_cash = 0
for i in range(0, 10087):
    if fine_df['state'][i] == "FL":
        if fine_df['cash_fine'][i] != '0':
            fl_cash += 1
        else:
            fl_online += 1


tx_online= 0
tx_cash = 0
for i in range(0, 10087):
    if fine_df['state'][i] == "TX":
        if fine_df['cash_fine'][i] != '0':
            tx_cash += 1
        else:
            tx_online += 1

pa_online= 0
pa_cash = 0
for i in range(0, 10087):
    if fine_df['state'][i] == "PA":
        if fine_df['cash_fine'][i] != '0':
            pa_cash += 1
        else:
            pa_online += 1

In [81]:
states = ['CA','NY','FL','TX','PA']
state_cash = []
state_cash.append(ca_cash)
state_cash.append(ny_cash)
state_cash.append(fl_cash)
state_cash.append(tx_cash)
state_cash.append(pa_cash)
state_cash

[524, 395, 337, 273, 252]

In [82]:
state_onl = []
state_onl.append(ca_online)
state_onl.append(ny_online)
state_onl.append(fl_online)
state_onl.append(tx_online)
state_onl.append(pa_online)
state_onl

[648, 438, 401, 351, 281]

In [83]:
states = ['CA','NY','FL','TX','PA']

fig = go.Figure(data=[
    go.Bar(name='Online', x=states, y=state_onl),
    go.Bar(name='Cash', x=states, y=state_cash)
])
# Change the bar mode
fig.update_layout(barmode='group')
fig.show()